File: solution/task1/python/libclimate.py
import math
from decimal import Decimal
import random
import pprint as pprint


class Libclimate:


    def gen_humidity (self, dew_point, temperature):
        humidity = 100.00 * ((611 * math.exp(5423 * ((1 / 273)) - (1 / (dew_point + 273)))) / (611 * math.exp(5423 * ((1 / 273)-(1 / temperature)))))
        print('humidity' , str(humidity))
        return(humidity)
       
          
    def gen_pressure(self, temperature):
         pressure = 101325 * math.exp(((0.00 - 9.81) * 0.0289644 * (200))/(8.31444598 * (temperature + 273)))
         print('pressure' , str(pressure))
         return (pressure)
         
         

    def gen_wind_speed(self, longitude):
        wind_speed = ((0.0001) * longitude)
        print('wind_speed' , str(wind_speed))
        return ( wind_speed)
       
        
        
    def dew_point_calculator(self, relative_humidity):
        dew_point = (100.00 - relative_humidity) / 5
        print('dew_point' , str(dew_point))
        return (dew_point)
        
        

    def relative_humidity_calculator(self):
        upper_hum = 75.00
        lower_hum = 55.00
        relative_humidity = random.uniform(lower_hum, upper_hum)
        print ('relative_humidity' , str(relative_humidity))
        return(relative_humidity)
        

    def temperature_generator(self):
        upper_temp = 42.00
        lower_temp = 15.00
        temperature = random.uniform(lower_temp, upper_temp)
        print('temperature' , str(temperature))
        return (temperature)
        
        
    def longitude_generator(self):
        upper_l = 98.00
        lower_l = 102.05
        longitude = random.uniform(lower_l, upper_l)
        print('longitude' , str(longitude))
        return (longitude)

File: solution/task1/cpp/lib/libclimate.cc
#include <math.h>
#include <cstdlib>
#include <ctime>

#include "libclimate.hpp"

float Libclimate::gen_humidity(float dew_point, float temperature)
{
    float humidity = 100.00 * ((611 * expf(5423 * ((1 / 273)) - (1 / (dew_point + 273)))) / (611 * expf(5423 * ((1 / 273)-(1 / temperature)))));
    return humidity;
}

float Libclimate::gen_pressure(float temperature)
{
    float pressure = 101325 * expf(((0.00 - 9.81) * 0.0289644 * (200))/(8.31444598 * (temperature + 273)));
    return pressure;
}

float Libclimate::gen_wind_speed(float longitude)
{
    float wind_speed = ((0.0001) * longitude);
    return wind_speed;
}

float Libclimate::dew_point_calculator(float relative_humidity)
{
    float dew_point = (100.00 - relative_humidity) / 5;
    return dew_point;
}

float Libclimate::relative_humidity_calculator(void)
{
    srand((unsigned)time(0));
    float upper_hum = 75.00;
    float lower_hum = 55.00;
    float humidity_range = (upper_hum - lower_hum) + 1;
    float relative_humidity = lower_hum + (float)(humidity_range * rand() / (RAND_MAX + 1.00));
    return relative_humidity;
}

float Libclimate::temperature_generator(void)
{
    srand((unsigned)time(0));
    float upper_temp = 42.00;
    float lower_temp = 15.00;
    float temp_range = (upper_temp - lower_temp) + 1;
    float temperature = lower_temp + (float)(temp_range * rand() / (RAND_MAX + 1.00));
    return temperature;
}

float Libclimate::longitude_generator(void)
{
    srand((unsigned)time(0));
    float upper_l = 102.05;
    float lower_l = 98.00;
    float l_range = (upper_l - lower_l) + 1;
    float longitude = lower_l + (float)(l_range * rand() / (RAND_MAX + 1.00));
    return longitude;
}